sdks/java/core/src/test/java/org/apache/beam/sdk/util/IOChannelUtilsTest.java

```diff
@@ -80,6 +80,13 @@ public void testLargeShardCount() {
   }
 
   @Test
+  public void testFilePrefix() throws Exception {
+    IOChannelUtils.getFactory("file://tmp");
+    IOChannelUtils.getFactory("file:/tmp");
+    IOChannelUtils.getFactory("file:tmp");
```

Comment:
I think there's only the one case w.r.t. parsing; If the path starts with `file://` (exactly), that prefix can be dropped.

If you resolve the three paths above appropriately, the following behavior should be observed:
The file `tmp` in the current working directory (i.e. `$PWD/tmp`)
The file `tmp` in the directory `file:/` in the current working directory (i.e. `$PWD/file:/tmp`)
The file `file:tmp` in the current working directory (i.e. `$PWD/file:tmp`)

Notably as well, these are all of the expected behaviors; the latter two test cases above **do not** have a scheme, and are therefore defaulted to the `FileIOChannelFactory`

Additionally, this should test that an absolute path functions as expected (e.g. `file:///tmp/foo/bar`, which is equivalent to `/tmp/foo/bar`) - Generally, along with the tests that exercise the channel factories, these tests should also compare to an expected scheme-less path to behave identically to the absolute (incl. scheme) path.